libatalk/adouble/ad_attr.c

```diff
@@ -98,71 +96,59 @@ int ad_setattr(const struct adouble *ad, const uint16_t attribute)
 /* --------------
  * save file/folder ID in AppleDoubleV2 netatalk private parameters
  * return 1 if resource fork has been modified
- * return -1 on error.
  */
 int ad_setid (struct adouble *adp, const dev_t dev, const ino_t ino , const uint32_t id, const cnid_t did, const void *stamp)
 {
     EC_INIT;
     uint32_t tmp;
-    char *ade = NULL;
     ssize_t id_len = -1, dev_len = -1, ino_len = -1, did_len = -1, syn_len = -1;
 
     id_len = ad_getentrylen(adp, ADEID_PRIVID);
     ad_setentrylen( adp, ADEID_PRIVID, sizeof(id));
     tmp = id;
     if (adp->ad_vers == AD_VERSION_EA)
         tmp = htonl(tmp);
-
-    ade = ad_entry(adp, ADEID_PRIVID);
-    if (ade == NULL) {
-        LOG(log_warning, logtype_ad, "ad_setid(%s): failed to set ADEID_PRIVID", adp->ad_name);
+    if (!ad_entry(adp, ADEID_PRIVID)) {
+        LOG(log_debug, logtype_default, "ad_setid(%s): invalid PRIVID", adp->ad_name);
         EC_FAIL;
     }
-    memcpy(ade, &tmp, sizeof(tmp));
+    memcpy(ad_entry( adp, ADEID_PRIVID ), &tmp, sizeof(tmp));
 
     dev_len = ad_getentrylen(adp, ADEID_PRIVDEV);
     ad_setentrylen( adp, ADEID_PRIVDEV, sizeof(dev_t));
-    ade = ad_entry(adp, ADEID_PRIVDEV);
-    if (ade == NULL) {
-        LOG(log_warning, logtype_ad, "ad_setid(%s): failed to set ADEID_PRIVDEV", adp->ad_name);
+    if (!ad_entry(adp, ADEID_PRIVDEV)) {
+        LOG(log_debug, logtype_default, "ad_setid(%s): invalid PRIVDEV", adp->ad_name);
         EC_FAIL;
     }
-
     if ((adp->ad_options & ADVOL_NODEV)) {
-        memset(ade, 0, sizeof(dev_t));
+        memset(ad_entry( adp, ADEID_PRIVDEV ), 0, sizeof(dev_t));
     } else {
-        memcpy(ade, &dev, sizeof(dev_t));
+        memcpy(ad_entry( adp, ADEID_PRIVDEV ), &dev, sizeof(dev_t));
     }
 
     ino_len = ad_getentrylen(adp, ADEID_PRIVINO);
     ad_setentrylen( adp, ADEID_PRIVINO, sizeof(ino_t));
-    ade = ad_entry(adp, ADEID_PRIVINO);
-    if (ade == NULL) {
-        LOG(log_warning, logtype_ad, "ad_setid(%s): failed to set ADEID_PRIVINO", adp->ad_name);
+    if (!ad_entry(adp, ADEID_PRIVINO)) {
+        LOG(log_debug, logtype_default, "ad_setid(%s): invalid PRIVINO", adp->ad_name);
         EC_FAIL;
     }
-    memcpy(ade, &ino, sizeof(ino_t));
+    memcpy(ad_entry( adp, ADEID_PRIVINO ), &ino, sizeof(ino_t));
 
-    if (adp->ad_vers != AD_VERSION_EA) {
```

Comment:
Interesting that the check for EA style metadata was removed here. Otherwise the block’s logic seems unchanged. Is DID a valid concept for EA metadata? (I’m asking out of curiosity)